frontend/src/api/tus.js

```diff
@@ -0,0 +1,158 @@
+import * as tus from "tus-js-client";
+import { tusEndpoint } from "@/utils/constants";
+import store from "@/store";
+import { removePrefix } from "./utils";
+
+// Make following configurable by envs?
+export const chunkSize = 20 * 1000 * 1000;
+const parallelUploads = 3;
+const retryDelays = [0, 3000, 5000, 10000, 20000];
+
+export async function upload(url, content = "", overwrite = false, onupload) {
+  return new Promise((resolve, reject) => {
+    var upload = new tus.Upload(content, {
+      httpStack: new CustomHTTPStack(),
+      endpoint: tusEndpoint,
+      chunkSize: chunkSize,
+      retryDelays: retryDelays,
+      parallelUploads: parallelUploads,
+      metadata: {
+        filename: content.name,
+        filetype: content.type,
+        overwrite: overwrite,
+        destination: removePrefix(url),
```

Comment:
I discovered `destination` haven't decode url.
If an user want to upload a file, maybe a file named `[ABC].txt` or something else.
It will become `%5BABC%5D.txt` after upload, not original filename `[ABC].txt`.

Reply:
Thanks for reviewing and testing my PR. You are right, I'm not correctly decoding the URL encoded metadata values.
I've fixed this issue so that files with special characters can now be uploaded.